models.py

```diff
@@ -251,20 +251,12 @@ def get_default_impact_title(self):
         )
         return layer_name
 
-    _impact_function_list = []
-
-    @classmethod
-    def impact_function_list(cls):
-        if not cls._impact_function_list:
-            from geosafe.tasks.headless.analysis import filter_impact_function
-            cls._impact_function_list = filter_impact_function.delay().get()
-        return cls._impact_function_list
-
     def impact_function_name(self):
-        for i in self.impact_function_list():
-            if i['id'] == self.impact_function_id:
-                return i['name']
-        return ''
+        # Set impact function name from provenance data
+        from geosafe.tasks.headless.analysis import get_keywords
+        from geosafe.helpers.utils import get_layer_path
+        keywords = get_keywords.delay(get_layer_path(self.impact_layer)).get()
+        return keywords['provenance_data']['impact_function_name']
```

Comment:
I don't think we still need this?
What is it for?

Reply:
it's only for a view in analysis-list, I think it's still fine to show the name of the IF there